src/inpage/index.js

```diff
@@ -18,8 +19,13 @@ const metamaskStream = new ReactNativePostMessageStream({
 initializeProvider({
   connectionStream: metamaskStream,
   shouldSendMetadata: false,
+  providerInfo: {
+    uuid: uuid(),
+    name: "MetaMask Mobile",
```

Comment:
There is also mobile Flask. How do we account for this?

Reply:
I'm not sure what build time options we have on mobile.

The bundled contentpage and inpage scripts get consumed in these places in mobile:

* https://github.com/MetaMask/metamask-mobile/blob/2609bb59cea4985a5101665910505c6f3a95002b/scripts/build.sh#L112
* https://github.com/MetaMask/metamask-mobile/blob/2609bb59cea4985a5101665910505c6f3a95002b/app/core/EntryScriptWeb3.js#L10
* https://github.com/MetaMask/metamask-mobile/blob/2609bb59cea4985a5101665910505c6f3a95002b/app/components/Views/Root/index.js#L49C4-L49C4
* https://github.com/MetaMask/metamask-mobile/blob/2609bb59cea4985a5101665910505c6f3a95002b/app/components/Views/BrowserTab/index.js#L593


We already use [this pattern in mobile-provider](https://github.com/MetaMask/mobile-provider/blob/801970155a920590a8214ea425b4d3e264fddf20/src/inpage/index.js#L24-L32) of leveraging `window` for triggering a func in inpage from contentscript. We could perhaps write to the window property? idk

Reply:
@sethkfman would love any thoughts you may have on this 🙏

Reply:
@jiexi I would maybe reference how the Flask branch is setup [here](https://github.com/MetaMask/metamask-mobile/blob/1bc54c11f2c91ca78d0b7f5d63318aa24f4ad38b/scripts/build.sh#L128). This is not included in `main`. 

Can you elaborate more on your use case to make sure we are solving this correctly?

Reply:
In order to announce the correct provider to the dapp, we need:
* icon
  * data uri of regular/flask logo svg
* name
  * MetaMask' or 'MetaMask Flask'
* rdns identifier
  * 'io.metamask' or 'io.metamask.flask'
  
That data would need to be provided to the `initializeProvider()` call [here in mobile-provider](https://github.com/MetaMask/mobile-provider/blob/main/src/inpage/index.js#L18). Problem is that [metamask-mobile consumes](https://github.com/MetaMask/metamask-mobile/blob/744def8b673d67581575c54c736d7c64f365d19c/app/core/EntryScriptWeb3.js#L10) the built and minified bundle of mobile-provider. This doesn't give us a easy way to pass values back into the `initializeProvider()` call before it is ultimately [injected into a page](https://github.com/MetaMask/metamask-mobile/blob/982a509f86a372b53dd405b4df2762518defa467/app/components/Views/BrowserTab/index.js#L1488C39-L1488C39) in the mobile browser.

One thing we can do (which was my initial approach for extension before figuring out how to better utilize the build system) is to put placeholder strings into mobile-provider that allow us to do a search and replace in metamask-mobile. That would mean mobile-provider `initializeProvider()` call would look something like:
```
initializeProvider({
  connectionStream: metamaskStream,
  shouldSendMetadata: false,
  providerInfo: {
    uuid: "UUID_PLACEHOLDER",
    name: "NAME_PLACEHOLDER",
    icon: "ICON_DATA_URI_PLACEHOLDER",
    rdns: "RDNS_PLACEHOLDER"
  }
});
```

The bundling and minification would likely leave the placeholder strings themselves intact so that in [EntryScriptWeb3.js on metamask-mobile](https://github.com/MetaMask/metamask-mobile/blob/744def8b673d67581575c54c736d7c64f365d19c/app/core/EntryScriptWeb3.js#L10) we could do something like:
```
...
  async init() {
    const script = Device.isIos()
      ? await RNFS.readFile(
          `${RNFS.MainBundlePath}/InpageBridgeWeb3.js`,
          'utf8',
        )
      : await RNFS.readFileAssets(`InpageBridgeWeb3.js`);

    this.entryScriptWeb3 = script
      .replace('UUID_PLACEHOLDER', uuid())
      .replace('NAME_PLACEHOLDER', isFlask? 'MetaMask Flask' : 'MetaMask')
      .replace('ICON_DATA_URI_PLACEHOLDER', '...')
      .replace('RDNS_PLACEHOLDER', isFlask? 'io.metamask.flask' : 'io.metamask')      

    return this.entryScriptWeb3;
  },
...
```

The approach above can be brittle though. Is there another way you see we could tackle this? It may simply be that mobile-provider needs to be pulled into metamask-mobile and built there instead

Reply:
Bring @metamask/mobile-provider back into metamask-mobile
https://github.com/MetaMask/metamask-mobile/pull/7494